### ISSUE TITLE
Etcd v3 Range Query load-balancing between nodes

### ISSUE THREAD

**Issue by prune998** [etcd-io/etcd#9499]
# versions
Etcd 3.2.17 started on GKE Kubernetes 1.9.4 cluster, created using etcd-operator. 3 node cluster on dedicated K8s nodes in the same zone, using SSD local disks.

# Description
We are monitoring the calls to Etcd cluster using Prom metrics `etcd_debugging_mvcc_xxx_total` where xxx is `range`, `put`, `delete`...

Looking at the graphs, is sounds that the range queries are unbalanced between the nodes : 

![image](https://user-images.githubusercontent.com/1110398/38034733-e88cd1c2-3270-11e8-967b-e25aa3691cd7.png)

This is worse for range, but is also true for other kind of queries. Ex, one node is processing far less `put` requests than the two others.

# expected result
We were expecting to have the queries spread equally between the nodes.

# code sample
This is the clientv3 code we are using to init etcd : 
```
	grpc_prometheus.EnableHandlingTimeHistogram()
	DialOptions := []grpc.DialOption{
		grpc.WithInsecure(),
		grpc.WithUnaryInterceptor(grpc_prometheus.UnaryClientInterceptor),
		grpc.WithStreamInterceptor(grpc_prometheus.StreamClientInterceptor),
	}
	cli, err := clientv3.New(clientv3.Config{
		Endpoints:   etcdAddrList,
		DialOptions: DialOptions,
	})
	if err != nil {
		log.Fatal(err)
	}
	defer cli.Close()
```

# discussion
I can't find a good documentation explaining how the Etcd Client is supposed to spread the load between nodes and if some type of queries have to be made to a specific node (master ?) 
Any pointer welcome

**Comment by gyuho** [comment]
Currently, we don't support balancing *loads* (it's in our roadmap). Balancer only supports failover. And sends loads only to the pinned endpoint.

**Comment by prune998** [comment]
Thanks for the answer. I still don't understand why one node is receiving more requests, I'm investigating.
Also, I would be interrested for a pointer on how to use `istio auto-inject sidecar` to the etcd-operator pods. I tried just adding the annotation without luck.. 🔨

**Comment by gyuho** [comment]
Jut an update: We've decided to go with round robin, with new clientv3 balancer (will be released with v3.4). For every request (given that sent from one shared client), loads will be balanced between multiple nodes.
